### PR TITLE
KAFKA-14588 [3/N] ConfigCommandTest rewritten in java

### PR DESCRIPTION
**Pull request by nizhikov**
This is third part of https://github.com/apache/kafka/pull/15417 refactoring.
PR intention - split big PR in parts to simplify review.
PR contains first part of ConfigCommandTest rewritten in java

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by nizhikov** [comment]
Parent PR is #15839

**Comment by nizhikov** [comment]
Hello @chia7712 

This PR is first part of rewriting `ConfigCommandTest` in java.
As you may see, test cases checks different variation of `ConfigCommand` options both - zk and kraft.
No broker started during the test.
I doubt we need to use `ClusterTest` here.

Can you, please, take a look?

**Comment by nizhikov** [comment]
CI looks OK for me. PR ready for review.

File: core/src/test/java/kafka/admin/JConfigCommandTest.java
Comment (chia7712) [inline core/src/test/java/kafka/admin/JConfigCommandTest.java]: maybe we can use java version?

File: core/src/test/java/kafka/admin/JConfigCommandTest.java
Comment (chia7712) [inline core/src/test/java/kafka/admin/JConfigCommandTest.java]: How about rename it to `ConfigCommandUnitTest`

**Review by chia7712 — COMMENTED** [review]
@nizhikov overall LGTM. two small comments left.

File: core/src/test/java/kafka/admin/JConfigCommandTest.java
Comment (nizhikov) [reply core/src/test/java/kafka/admin/JConfigCommandTest.java]: renamed

File: core/src/test/java/kafka/admin/JConfigCommandTest.java
Comment (nizhikov) [reply core/src/test/java/kafka/admin/JConfigCommandTest.java]: Yes we can. fixed.

**Comment by nizhikov** [comment]
CI looks OK 

@chia7712 Do you have any other comments?

File: core/src/test/java/kafka/admin/ConfigCommandUnitTest.java
Comment (chia7712) [inline core/src/test/java/kafka/admin/ConfigCommandUnitTest.java:370]: Could we replace those scala collection by java collection in this PR? As we are in JDK8+, most lambda functions should work in Java. In order to reduce the size of PR, we rewrite only production code related to this PR. WDYT?

Comment (nizhikov) [reply core/src/test/java/kafka/admin/ConfigCommandUnitTest.java:370]: Are you suggesting make `ConfigCommandOptions#entityTypes` return `java.util.List` instead of scala List?
I doubt it will make PR more clean, because `toSeq` and `seq` calls are pretty compact and self explaining.

We have full java version of Command waiting for a review after tests, also.
So, do we really need some intermediate steps?

Comment (chia7712) [reply core/src/test/java/kafka/admin/ConfigCommandUnitTest.java:370]: > We have full java version of Command waiting for a review after tests, also.
So, do we really need some intermediate steps?

I try to avoid large changes when we do "full java version of Command". Changing all collections from Scala to Java could result in another large "rewrite" for those tests. If those tests keep being "rewrite", what the point of separating those "smaller" patches?

I'm ok to keep current version if you feel above concern is false :)

Comment (nizhikov) [reply core/src/test/java/kafka/admin/ConfigCommandUnitTest.java:370]: > Changing all collections from Scala to Java could result in another large "rewrite" for those tests.

First of all - I understand your point and agree with it.
I'm absolutely +1 for making patches as small and independent as possible to avoid any possible bugs during scala -> java migration.
Let me try to change collections from scala to java as another change before review of the command itself.
After we merge all tests in java.

Will it work for you?

> If those tests keep being "rewrite", what the point of separating those "smaller" patches?

To separate two reviews and eye checks - one for test itself - to make sure all checks are same in java and scala and I not removed some cases accidently.

Change of collection types must be similar to each other, so it will be easy to review separately.    

> I'm ok to keep current version if you feel above concern is false :)

Let's keep current version, please.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**
